src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -498,40 +498,110 @@ bool SketcherGui::isSketchInEdit(Gui::Document* doc)
-            if (static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit())->getSketchMode()
-                == ViewProviderSketch::STATUS_NONE) {
-                if (!actsOnSelection) {
-                    return true;
+            selectionHasGeo = false;
+            selectionHasBSpline = false;
+            selectionHasOnlyBsplines = true;
+            selectionHasOnlyElipses = true;
+            selectionHasOnlyPoints = true;
+            selectionHasConstraint = false;
+
+            std::vector<Gui::SelectionObject> selection;
+            selection =
+                Gui::Selection().getSelectionEx(0, Sketcher::SketchObject::getClassTypeId());
+
+            if (selection.empty()) {
+                return false;
+            }
+
+            auto* Obj = static_cast<Sketcher::SketchObject*>(selection[0].getObject());
+            const std::vector<std::string>& subNames = selection[0].getSubNames();
+            if (!subNames.empty()) {
+
+                for (auto& name : subNames) {
+                    if (name.size() > 4 && name.substr(0, 4) == "Edge") {
+                        selectionHasGeo = true;
+                        selectionHasOnlyPoints = false;
+                        int geoId = std::atoi(name.substr(4, 4000).c_str()) - 1;
+                        if (isBSplineCurve(*Obj->getGeometry(geoId))) {
+                            selectionHasBSpline = true;
+                            selectionHasOnlyElipses = false;
+                        }
+                        else if (isArcOfEllipse(*Obj->getGeometry(geoId))
+                                 || isEllipse(*Obj->getGeometry(geoId))) {
+                            selectionHasOnlyBsplines = false;
+                        }
+                        else {
+                            selectionHasOnlyElipses = false;
+                            selectionHasOnlyBsplines = false;
+                        }
+                    }
+                    else if (name.size() > 12 && name.substr(0, 12) == "ExternalEdge") {
+                        selectionHasGeo = true;
+                        selectionHasOnlyBsplines = false;
+                        selectionHasOnlyElipses = false;
+                        selectionHasOnlyPoints = false;
+                    }
+                    else if (name.size() > 6 && name.substr(0, 6) == "Vertex") {
+                        int VtId = std::atoi(name.substr(6, 4000).c_str()) - 1;
+                        int geoId;
+                        Sketcher::PointPos PosId;
+                        selectionHasOnlyElipses = false;
+                        selectionHasOnlyBsplines = false;
+                        Obj->getGeoVertexIndex(VtId, geoId, PosId);
+                        selectionHasGeo = true;
+                    }
+                    else if (name.size() > 10 && name.substr(0, 10) == "Constraint") {
+                        selectionHasConstraint = true;
+                    }
```

Comment:
This computation really should be extracted to its own function returning some kind of struct / bitset. Then this function can be debounced and throttled properly.